code/modules/surgery/organs/heart.dm

```diff
@@ -62,6 +62,11 @@
 
 /obj/item/organ/heart/on_life()
 	..()
+
+	// If the owner doesn't need a heart, we don't need to do anything with it.
+	if(!owner.needs_heart())
+		return
```

Comment:
How does this interact with cursed and/or robotic hearts? It looks like this will make upgraded robotic hearts not inject plasmamen with chems.

Reply:
No change to cursed_heart, it uses snowflake on_life. I looked over it and decided that I like the idea that the cursed heart ignores modifiers except from species that lack blood. It IS cursed afterall.

The cybernetic heart calls the parent on_life() but doesn't do anything with the return value and operates as normal. I could and should probably refactor this a bit, however. I don't like the idea that implanting a useless cybernetic heart into a species or mob that doesn't use it will allow them to gain benefits from it. Seems like a consistency issue.

Reply:
> I don't like the idea that implanting a useless cybernetic heart into a species or mob that doesn't use it will allow them to gain benefits from it. Seems like a consistency issue.

I'd argue the opposite, actually- in my headcanon, the  chem injection device is sort of strapped onto the robotic heart itself. Even though the heart is useless for a plasmaman, the chem injection device is not, as plasmamen have livers and can process chems.